### PR TITLE
1559

### PR DESCRIPTION
**Pull request by tkstanczak**

### CODE REVIEW

File: src/Nethermind/Nethermind.Blockchain/Comparers/GasPriceTxComparer.cs [hunk 1]
```diff
@@ -35,6 +34,7 @@ public GasPriceTxComparer(IBlockTree blockTree, ISpecProvider specProvider)
         
         public int Compare(Transaction? x, Transaction? y)
         {
+            // When we don't know next block base fee, we're getting it from the current Head. That should be great approximation
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Blockchain/Comparers/GasPriceTxComparer.cs:37]: I would like to learn more about it?

Comment (tkstanczak) [reply src/Nethermind/Nethermind.Blockchain/Comparers/GasPriceTxComparer.cs:37]: we can calculate it from head, right?

File: src/Nethermind/Nethermind.TxPool/TxPool.cs [hunk 2]
```diff
@@ -317,7 +317,7 @@ private AddTxResult AddCore(Transaction tx, bool isPersistentBroadcast, bool isR
                     _logger.Trace($"Skipped adding transaction {tx.ToString("  ")}, cost overflow.");
                 return AddTxResult.BalanceOverflow;
             }
-            else if ((account?.Balance ?? UInt256.Zero) < cost)
+            else if ((account?.Balance ?? UInt256.Zero) < cost && !tx.IsEip1559)
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.TxPool/TxPool.cs:320]: I would like to understand this change in details

File: src/Nethermind/Nethermind.Blockchain/Comparers/GasPriceTxComparer.cs (hunk 1 above)
Comment (MarekM25) [reply src/Nethermind/Nethermind.Blockchain/Comparers/GasPriceTxComparer.cs]: Yes. I changed that comment a bit.
That is how I see it:
When we're adding Tx to TxPool we don't know the base fee of the block in which transaction will be added. What we can do is get a base fee from the head in two way:
1. head.Header.BaseFee - the current one
2. BlockHeader.CalculateBaseFee(head, spec) - but I am not sure if it is worth recalculating a base fee here

File: src/Nethermind/Nethermind.TxPool/TxPool.cs (hunk 2 above)
Comment (MarekM25) [reply src/Nethermind/Nethermind.TxPool/TxPool.cs:320]: That was a temporary ugly hack. Changed to calculate the balance properly in the context of 1559. However, I will add a few more checks to TxPool.

**Pull request closed without merging**